diff --git a/meson.build b/meson.build
index a42897f..65a7f59 100755
--- a/meson.build
+++ b/meson.build
@@ -4,7 +4,11 @@ project('CrayonBall', 'c',
 )
 
 cc = meson.get_compiler('c')
-ruby_dir = 'ruby-1.8.7'
+if target_machine.cpu_family() == 'x86_64'
+	ruby_dir = '/app/lib/ruby/1.8/x86_64-linux'
+elif target_machine.cpu_family() == 'aarch64'
+	ruby_dir = '/app/lib/ruby/1.8/aarch64-linux'
+endif
 
 exe = executable('aerosol',
 	[
@@ -83,7 +87,7 @@ exe = executable('aerosol',
 		cc.find_library('OpenGL'),
 		cc.find_library('GLU'),
 		cc.find_library('openal'),
-		cc.find_library('ruby-static', dirs: join_paths(meson.current_source_dir(), ruby_dir)),
+		cc.find_library('ruby-static'),
 	],
 	c_args: [
 		'-Wno-unused-parameter',
